pkg/upgradeservice/preflight/preflight.go

```diff
@@ -144,12 +134,21 @@ func Run(app *apptypes.App, archiveDir string, sequence int64, registrySettings
 	return nil
 }
 
-func setPreflightResults(results *types.PreflightResults) error {
+func setPreflightResults(results *types.PreflightResults, runError error) error {
 	preflightData, err := getPreflightData()
 	if err != nil {
 		return errors.Wrap(err, "failed to get preflight data")
 	}
 	preflightData.Results = results
+	if runError != nil {
+		if preflightData.Results.Errors == nil {
```

Comment:
sorry to be nit picky, but to be consistent with the other implementation, let's update `results` before updating `preflightData.Results`. would probably look cleaner.